server/src/test/java/org/apache/iotdb/db/mpp/sql/plan/DistributionPlannerTest.java

```diff
@@ -219,6 +267,33 @@ private Analysis constructAnalysis() {
     dataPartition.setDataPartitionMap(dataPartitionMap);
 
     analysis.setDataPartitionInfo(dataPartition);
+
+    // construct schema partition
+    SchemaPartition schemaPartition = new SchemaPartition();
+    Map<String, Map<SeriesPartitionSlot, RegionReplicaSet>> schemaPartitionMap = new HashMap<>();
+
+    RegionReplicaSet schemaRegion1 =
+        new RegionReplicaSet(
+            new ConsensusGroupId(GroupType.SchemaRegion, 11),
+            Arrays.asList(
+                new DataNodeLocation(11, new Endpoint("192.0.1.1", 9000)),
+                new DataNodeLocation(12, new Endpoint("192.0.1.2", 9000))));
+    Map<SeriesPartitionSlot, RegionReplicaSet> schemaRegionMap = new HashMap<>();
+
+    RegionReplicaSet schemaRegion2 =
+        new RegionReplicaSet(
+            new ConsensusGroupId(GroupType.SchemaRegion, 21),
+            Arrays.asList(
+                new DataNodeLocation(21, new Endpoint("192.0.1.1", 9000)),
+                new DataNodeLocation(22, new Endpoint("192.0.1.2", 9000))));
+
+    schemaRegionMap.put(new SeriesPartitionSlot(1), schemaRegion1);
```

Comment:
Currently we use a mock method to calculate the DeviceGroupId from device name. That is, the DeviceGroupId is the length of device name. So... here the `SeriesPartitionSlot(1)` should be something like `SeriesPartitionSlot("d1".length)`